src/pymor/bindings/dunegdt.py

```diff
@@ -0,0 +1,227 @@
+
+        Parameters
+        ----------
+        impl
+            The actual vector from dune.xt.la, usually IstlVector.
+        """
+
+        def __init__(self, impl):
+            self.impl = impl
+
+        @classmethod
+        def from_instance(cls, instance):
+            return cls(instance.impl)
+
+        def _copy_data(self):
+            self.impl = self.impl.copy(True)
+
+        def _scal(self, alpha):
+            self.impl.scal(alpha)
+
+        def _axpy(self, alpha, x):
+            self.impl.axpy(alpha, x.impl)
+
+        def inner(self, other):
+            return self.impl.dot(other.impl)
+
+        def norm(self):
+            return self.impl.l2_norm()
+
+        def norm2(self):
+            return self.impl.l2_norm() ** 2
+
+        def sup_norm(self):
+            return self.impl.sup_norm()
+
+        def dofs(self, dof_indices):
+            impl = self.impl
+            return np.array([impl[i] for i in dof_indices])
+
+        def amax(self):
+            _amax = self.impl.amax()
+            return _amax[0], _amax[1]
+
+        def __add__(self, other):
+            return DuneXTVector(self.impl + other.impl)
+
+        def __iadd__(self, other):
+            self.impl += other.impl
+            return self
+
+        __radd__ = __add__
+
+        def __sub__(self, other):
+            return DuneXTVector(self.impl - other.impl)
+
+        def __isub__(self, other):
+            self.impl -= other.impl
+            return self
+
+        def __mul__(self, other):
+            return DuneXTVector(self.impl * other)
+
+        def __imul__(self, other):
+            self.impl *= other
+            return self
+
+        def __neg__(self):
+            return self * (-1)
+
+        def to_numpy(self, ensure_copy=False):
+            return np.array(self.impl, copy=ensure_copy)
+
+    class ComplexifiedDuneXTVector(ComplexifiedVector):
+        """Required for DuneXTVectorSpace, Usually not to be used directly."""
+
+        def amax(self):
+            if self.imag_part is None:
+                return self.real_part.amax()
+            else:
+                real = np.array(self.real_part.impl, copy=False)
+                imag = np.array(self.imag_part.impl, copy=False)
+                return NumpyVector(real + imag * 1j).amax()
+
+    class DuneXTVectorSpace(ComplexifiedListVectorSpace):
+        """A |VectorSpace| yielding DuneXTVector
+
+        Parameters
+        ----------
+        dim
+            Dimension of the |VectorSpace|, i.e., length of the resulting vectors.
+        vector_type
+            Type of the actual vector from dune.xt.la, usually IstlVector.
+        id
+            Identifier of the |VectorSpace|.
+        """
+
+        complexified_vector_type = ComplexifiedDuneXTVector
+
+        def __init__(self, dim, vector_type=IstlVector, id='STATE'):
+            self.__auto_init(locals())
+
+        def __eq__(self, other):
```

Comment:
This check should also include the `id` member.

Reply:
done in 889bb38